apis/config/v1beta1/configuration_types.go

```diff
@@ -344,6 +347,13 @@ type PodIntegrationOptions struct {
 	PodSelector *metav1.LabelSelector `json:"podSelector,omitempty"`
 }
 
+type DeploymentIntegrationOptions struct {
```

Comment:
@alculquicondor Is there any context the reason why we decided to introduce these selectors?

IIUS, we introduced the selectors for PodIntegration because the Pod will be created by various type controllers in spite of Kube official one like kube-controller-manager.

But, it seems that the Deployment use cases do not match the same reason with the Pod ones.
If we allow the Deployment integrations to add such selectors, adding selectors to all integrations like Job will be justified, right?

Reply:
You are right, we didn't really discuss this.

In fact, we took the decision not to add this for Jobs, in favor of using ValidatingAdmissionPolicy https://github.com/kubernetes-sigs/kueue/issues/2119

@mimowo @mwielgus do you have any additional justification for adding this?

/hold

Reply:
Or perhaps @vladikkuzn has some context that I missed